app/models/piece.rb

```diff
@@ -82,4 +86,36 @@ def move_to!(new_row, new_column)
       end
     end
   end
+
+  def moving_into_check?(new_row, new_column)
+    #We will switch to the new coordinates temporarily and see if we are in check!
+    current_row = self.position_row
+    current_column = self.position_column
+    self.update_attributes(position_row: new_row, position_column: new_column)
+
+    #define my king and all the pieces from the other team
```

Comment:
Yeah so the problem is that the check method only returns if the game is in check. The moving into check method is supposed to check if you are going to put yourself in check. I could rewrite the check method so that it returns which king is in check...but  I was not sure how to pass that information since the check method is supposed to return either true or false